src/eth/storage/postgres/postgres.rs

```diff
@@ -471,23 +484,35 @@ impl EthStorage for Postgres {
                         })
                         .map(|val| val.as_ref().to_owned());
 
-                    let mut tx = self.connection_pool.begin().await.context("failed to init transaction")?;
+                    let original_nonce: BigDecimal = original_nonce.unwrap_or_default().try_into()?;
+                    let original_balance: BigDecimal = original_balance.unwrap_or_default().try_into()?;
+
                     let block_number = i64::try_from(block.header.number).context("failed to convert block number")?;
-                    let balance = BigDecimal::try_from(balance)?;
-                    let nonce = BigDecimal::try_from(nonce)?;
 
-                    sqlx::query_file!(
+                    let account_result: PgQueryResult = sqlx::query_file!(
                         "src/eth/storage/postgres/queries/insert_account.sql",
                         change.address.as_ref(),
                         nonce,
                         balance,
                         bytecode,
-                        block_number
+                        block_number,
+                        original_nonce,
+                        original_balance
                     )
                     .execute(&mut *tx)
                     .await
                     .context("failed to insert account")?;
 
+                    if account_result.rows_affected() != 1 {
```

Comment:
can you create a function to represent this case? or at least add some comments

Reply:
Sure